allennlp_models/generation/dataset_readers/seq2seq.py

```diff
@@ -91,15 +97,17 @@ def __init__(
             or target_add_start_token
             or target_add_end_token
         ):
-            try:
-                self._start_token, self._end_token = self._source_tokenizer.tokenize(
-                    start_symbol + " " + end_symbol
-                )
-            except ValueError:
+            # Check that the tokenizer correctly appends the start and end tokens to
+            # the sequence without splitting them.
+            tokens = self._source_tokenizer.tokenize(start_symbol + " " + end_symbol)
+            start_token, end_token = tokens[0], tokens[-1]
```

Comment:
We should also catch the potential `IndexError` that could happen here.

Reply:
Okay, I wrapped it in a try/except. It's a little ugly, let me know if you see a cleaner way to check.